contracts/mirror_factory/src/contract.rs

```diff
@@ -328,20 +324,14 @@ pub fn reply(deps: DepsMut, env: Env, msg: Reply) -> StdResult<Response> {
 /// 2. Register asset to mint contract
 /// 3. Register asset and oracle feeder to oracle contract
 /// 4. Create terraswap pair through terraswap factory with `TerraswapCreationHook`
-pub fn token_creation_hook(
+pub fn token_creation_reply(
     deps: DepsMut,
     env: Env,
     asset_token: Addr,
     oracle_feeder: Addr,
+    params: Params,
 ) -> StdResult<Response> {
     let config: Config = read_config(deps.storage)?;
-
-    // If the param is not exists, it means there is no whitelist process in progress
-    let params: Params = match read_params(deps.storage) {
```

Comment:
We can remove this because the reply is always executed after the parent operation stores the params.